src/fhirtypes/common.ts

```diff
@@ -198,6 +198,20 @@ export function setPropertyOnInstance(
           if (current[key] == null) current[key] = {};
           current = current[key];
         } else {
+          // Check if the instance  already has the element being defined
+          if (current.hasOwnProperty([pathPart.base])) {
```

Comment:
I'm confused by the array syntax here.  Why `current.hasOwnProperty([pathPart.base])` and not `current.hasOwnProperty(pathPart.base)`?  Is it just a left-over of a previous version where maybe you previously had `current[pathpart.base]`?  Or is there some other new fancy syntax thing I don't know about?  After all, it seems to work -- so I'm inclined to think maybe it was intentional!

It's not a big deal, but we could be a little more specific here and check that it's actually an object:
```ts
if (current[pathPart.base] != null && typeof current[pathPart.base] === 'object') {
```

Reply:
Oh.  I see @mint-thompson already noted this (in addition to some things I missed).  Thanks, @mint-thompson!

Reply:
It's not intentional! It was a holdover from a previous implementation where I was checking if `current[pathPart.base]` was truthy rather than using the `hasOwnProperty()` function. Looking at it now, I'm actually somewhat surprised it still worked! I also think that checking that it's actually an object would be useful, I'll add that app in another commit.